src/check.c

```diff
@@ -400,16 +400,42 @@ usercheck(pwquality_settings_t *pwq, const char *new,
                 ++f;
         }
 
-        if (strstr(new, user) != NULL)
+        if (strstr(new, word) != NULL)
                 return 1;
 
-        dist = distance(new, user);
+        dist = distance(new, word);
         if (dist >= 0 && dist < PWQ_DEFAULT_DIFF_OK)
                 return 1;
 
         return 0;
 }
 
+static int
+usercheck(pwquality_settings_t *pwq, const char *new,
+          char *user)
+{
+        int i, userlen;
+        int rv = 0;
+        char* subuser = calloc(pwq->user_substr+1, sizeof(char));
```

Comment:
char *subuser....

Please also handle NULL return from calloc. This will however require changing the callers to recognize  PWQ_ERROR_MEM_ALLOC return.

Reply:
I tweaked the return in the wordcheck and usercheck to return the actual error code instead of a boolean value.  I think that should take care of it (worked for me).  Lemme know if that's what you had in mind. :)